pipeline/beam_tables.py

```diff
@@ -377,29 +395,21 @@ def _data_to_load(self,
     else:
       files_to_load = SCAN_FILES
 
-    # Both zipped and unzipped data to be read in
-    zipped_regex = self.bucket + scan_type + '/**/{0}.gz'
-    unzipped_regex = self.bucket + scan_type + '/**/{0}'
-
-    file_metadata = []
-    for file in files_to_load:
-      zipped_metadata = [
-          m.metadata_list for m in gcs.match([zipped_regex.format(file)])
-      ][0]
-      unzipped_metadata = [
-          m.metadata_list for m in gcs.match([unzipped_regex.format(file)])
-      ][0]
-      file_metadata += zipped_metadata + unzipped_metadata
+    # Filepath like `gs://firehook-scans/echo/**/*'
+    files_regex = f'{self.bucket}{scan_type}/**/*'
+    file_metadata = [m.metadata_list for m in gcs.match([files_regex])][0]
```

Comment:
Nice change. gcs.match is expensive!

In the other PR I suggest a solution without match. You can use [google.cloud.storage.client.Client.list_blobs](https://googleapis.dev/python/storage/latest/client.html#google.cloud.storage.client.Client.list_blobs) with a start offset. You can stop listing when the end date is reached, or specify the end offset. Also, it returns an iterator, so you can start outputting filenames to be processed right away *before* the full listing is completed. The parallelism is significant.

Furthermore, the listing work is proportional to the work that needs to be done. That's the approach I use for the OONI data, and is a lot more efficient.

Reply:
(Also partially responding to your comment in https://github.com/censoredplanet/censoredplanet-analysis/pull/94)

It's not terribly expensive. Currently the slowest version of this call we make is
`gcs.match('gs://firehook-scans/satellite/**/*')`
which matches 4310 files

That's basically equivalent to running 
`gsutil ls gs://firehook-scans/satellite/**/*`

If I time that call I get 
`1.45s user 0.21s system 85% cpu 1.933 total`

Which is nothing on a pipeline that's going to take several minutes to get loaded into dataflow and probably several hours to run. (Still nice not to have to do it 22 times though.)

How long does it take you to enumerate the OONI files?

Reply:
OONI was way worse because we had subdirectories for each hour of each day, and then at least a file per country and test type (Example filename: `raw/20210508/04/TR/webconnectivity/2021050804_TR_webconnectivity.n0.0.jsonl.gz`). Listing could take a few minutes.

This single date took 14.5s:
```
time aws --no-sign-request s3 ls s3://ooni-data-eu-fra/raw/20210526/ --recursive
```